libraries/logstash_service_base.rb

```diff
@@ -20,10 +21,24 @@ class LogstashServiceBase < Chef::Resource
     default_action :start
     allowed_actions :start, :stop, :restart
 
+    def self.provides(*args, **kwargs)
```

Comment:
I don't believe this is good way to manage init style. Chef has embedded features for dynamically manage "providers"